runtime/runtime/src/verifier.rs

```diff
@@ -133,49 +136,70 @@ pub(crate) fn validate_transaction_well_formed<'a>(
     current_protocol_version: ProtocolVersion,
 ) -> Result<(), InvalidTxError> {
     validate_transaction_actions(config, signed_tx, current_protocol_version)?;
-    ValidatedTransaction::check_valid_for_config(config, signed_tx)
+    ValidatedTransaction::check_valid_for_config(config, signed_tx, current_protocol_version)
 }
 
-/// Set new `signer` and `access_key` in `state_update`.
-///
-/// Note that this does not commit state changes to the `TrieUpdate`.
-pub fn set_tx_state_changes(
+pub fn set_tx_balance_changes(
     state_update: &mut TrieUpdate,
-    validated_tx: &ValidatedTransaction,
-    signer: &Account,
-    access_key: &AccessKey,
+    account_id: AccountId,
+    key: TransactionKeyRef,
+    payer: &TransactionPayer,
 ) {
-    let tx = validated_tx.to_tx();
-    set_access_key(state_update, tx.signer_id().clone(), tx.public_key().clone(), &access_key);
```

Comment:
note for reviewers: in case you are wondering what happened to `set_access_key` (aka updating nonce) here, I split the responsibility of this function to `set_tx_balance_changes` and `set_tx_none_changes`.